services/tone_analyzer/v3.js

```diff
@@ -0,0 +1,152 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+module.exports = function (RED) {
+  var watson = require('watson-developer-cloud');  
+  var cfenv = require('cfenv');
+
+  // Require the Cloud Foundry Module to pull credentials from bound service 
+  // If they are found then they are stored in sUsername and sPassword, as the 
+  // service credentials. This separation from sUsername and username to allow 
+  // the end user to modify the node credentials when the service is not bound.
+  // Otherwise, once set username would never get reset, resulting in a frustrated
+  // user who, when he errenously enters bad credentials, can't figure out why
+  // the edited ones are not being taken.
+
+  // Not ever used, and codeacy complains about it.
+
+  var username, password, sUsername, sPassword;
+
+  var service = cfenv.getAppEnv().getServiceCreds(/tone analyzer/i)
+
+  if (service) {
+    sUsername = service.username;
+    sPassword = service.password;
+  }
+
+  // Node RED Admin - fetch and set vcap services
+  RED.httpAdmin.get('/watson-tone-analyzer/vcap', function (req, res) {
+    res.json(service ? {bound_service: true} : null);
+  });
+
+
+  var processOnInput = function(msg, config, node) {
+      var message = '';
+
+    // added in this flag as codacy was complaing about the 
+    // cyclomatic complexity. Essentially the number of exit points. With
+    // this flag we can reduce that do two exits. One if there is an
+    // error detected and the second for normal processing
+    var errorDetected = false;
+    var isBuffer = false;
+
+      // Credentials are needed for each of the modes.
+      username = sUsername || node.credentials.username;
+      password = sPassword || node.credentials.password;      
+
+      if (!username || !password) {
+        message = 'Missing Tone Analyzer service credentials';
+        errorDetected = true;
+      }
+
+      if (!msg.payload) {
+        message = 'Missing property: msg.payload';
+        errorDetected = true;
+      } else {
+        var hasJSONmethod = (typeof msg.payload.toJSON === 'function') ;
+
+        if (hasJSONmethod === true) {
+          if (msg.payload.toJSON().type === 'Buffer') {
+            isBuffer = true;
+          }      
+        }      
+      }
+
+      // Payload (text to be analysed) must be a string (content is either raw string or Buffer)
+      if (typeof msg.payload !== 'string' &&  isBuffer !== true) {
+        errorDetected = true;
+        message = 'The payload must be either a string or a Buffer';
+      }
+
+      if (errorDetected) {
+        node.status({fill:'red', shape:'dot', text:message}); 
+        node.error(message, msg);
+        return;
+      }
+
+      var tones = msg.tones || config.tones;
+      var sentences = msg.sentences || config.sentences;
+      var contentType = msg.contentType || config.contentType
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 4 space characters but found 6.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2652666678/issues/source?bid=3225115&fileBranchId=3325196#l92)